src/pages/HomePage.tsx

```diff
@@ -1,24 +1,60 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
+import { DocumentData } from "firebase/firestore"
+import { RootStateOrAny, useDispatch, useSelector } from "react-redux";
+import { getUserFromDB } from "../redux/actionCreators";
 import Header from "../components/header/Header";
 import Footer from "../components/footer/Footer";
 import { Recepie } from "../components/recepie/Recepie";
-import avatar from "../assets/john-doe.png";
+import defaultAvatar from "../assets/default-avatar.jpg"
 import "./HomePage.css";
 
 function HomePage() {
+    const dispatch = useDispatch();
+    const id = useSelector((state: RootStateOrAny) => state.auth.authInfo.uid);
+    const email = useSelector((state: RootStateOrAny) => state.auth.authInfo.email);
+    const userDataFromDB = useSelector((state: RootStateOrAny) => state.user.userInfo);
+    const errorInfo = useSelector((state: RootStateOrAny) => state.user.errorInfo);
+
+    const [userData, setUserData] = useState<DocumentData>({});
+    const [error, setError] = useState("");
+
+    useEffect(() => {
+        dispatch(getUserFromDB({ id }));
+    }, []);
+
+    useEffect(() => {
+        let data = userDataFromDB;
+        if (!userDataFromDB.name) {
+            data = { ...data, name: email }
+        }
+        if (!userDataFromDB.avatar) {
+            data = { ...data, avatar: defaultAvatar }
+        }
```

Comment:
I believe better approach here is move this logic to saga and save in state already mapped model